test_data/rules/unused_top_members_in_executable_libraries.dart

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `dart test -N unused_top_members_in_executable_libraries`
```

Comment:
Because I'm not familiar with `traverseNodesInDFS`, and therefore don't trust it implicitly :), I'd love to see some interesting cases like:

```dart
class C {
  const C();
}

void f<T extends C> {} // Usage in type bound

void Function(C c) f2() {} // Usage in Function type

void f3([Object p = C()]) {} // Usage in default value

@C() // Usage in annotation
void f4() {}

extension on Future<C> {} // Usage in type parameter in extension `on` clause.

Reply:
I added those tests

Reply:
🎉 Thanks!